lib/Doctrine/DBAL/Driver/PDOStatement.php

```diff
@@ -145,9 +200,51 @@ public function fetchAll($fetchMode = null, $fetchArgument = null, $ctorArgs = n
     public function fetchColumn($columnIndex = 0)
     {
         try {
-            return parent::fetchColumn($columnIndex);
+            return $this->stmt->fetchColumn($columnIndex);
         } catch (\PDOException $exception) {
             throw new PDOException($exception);
         }
     }
+
+    /**
+     * Converts DBAL parameter type to PDO parameter type
+     *
+     * @param int $type Parameter type
+     * @return int
+     */
+    private function convertParamType(int $type) : int
+    {
+        if ( ! isset(self::$paramTypeMap[$type])) {
+            throw new \InvalidArgumentException('Invalid parameter type: ' . $type);
+        }
+
+        return self::$paramTypeMap[$type];
+    }
+
+    /**
+     * Converts DBAL fetch mode to PDO fetch mode
+     *
+     * @param int|null $fetchMode Fetch mode
+     * @return int|null
+     */
+    private function convertFetchMode(?int $fetchMode) : ?int
+    {
+        if ($fetchMode === null) {
```

Comment:
Better move this out of here and drop nullability - null isn't a real fetch mode (or is it?).

Reply:
It means "use the default fetch mode set by `setFetchMode()`".